src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIService.java

```diff
@@ -140,7 +146,10 @@ private void setupGitLabCIConfiguration(VcsRepositoryUri repositoryUri, Programm
 
         try {
             // TODO: Reduce the number of API calls
-
+            if (exercise.getBuildConfig() == null || !Hibernate.isInitialized(exercise.getBuildConfig())) {
```

Comment:
Why don't you make sure that the programming exercise is already loaded properly with the configuration before it is passed into this method?

Reply:
The original value could come from different parts of the code, where the configration isn't needed in much of the process